src/sqlfluff/core/config.py

```diff
@@ -267,13 +283,30 @@ def _incorporate_vals(ctx: dict, vals: List[Tuple[Tuple[str, ...], Any]]) -> dic
             r[n] = v
         return ctx
 
-    def load_default_config_file(self, file_dir: str, file_name: str) -> dict:
+    @staticmethod
+    def _validate_configs(configs: Iterable[ConfigElemType], file_path):
+        """Validate config elements against deprecated list."""
+        for k, _ in configs:
+            if k in DEPRECATED_CONFIGS:
+                formatted_key = ":".join(k)
+                raise SQLFluffUserError(
+                    f"Config file {file_path} set deprecated config "
```

Comment:
The typical pattern I've seen for deprecation is to warn the user, then in a future release to remove it or raise an error. Wondering if we want to consider changing the behavior and/or use a term other than "deprecate".

Reply:
Hmmmm - good idea. I'm going to change the wording. I think the more "brutal" approach is good because otherwise users may get strange linting results which they look at before any raised warnings.